### ISSUE TITLE
Timestep Bias and Embedder Performance

### ISSUE THREAD

**Issue by dreaming-huang** [ByteDance-Seed/Bagel#199]
Regarding the timestep, I noticed that during the sampling of the dataset, the timestep is set as timestep = np.random.randn(). Then, during training, it is updated as:
`packed_timesteps = torch.sigmoid(packed_timesteps)`
Doesn't this mean that during training, most of the time is spent on timesteps closer to the middle of the range?

Additionally, I found that the time embedder is implemented as:
```
# --------------------------------------------------------
# TimestepEmbedder
# Reference:
# DiT: https://github.com/facebookresearch/DiT/blob/main/models.py
# --------------------------------------------------------
class TimestepEmbedder(nn.Module):
```
In the TimestepEmbedder, without modification, the output **t_freq** for **0 < t < 1** is extremely similar (although the subsequent MLP slightly separates them). I would like to discuss whether this could affect the performance.

**Comment by Andy1621** [comment]
1. Yes. Most of the timesteps are closer to the middle of the range. This is a common strategy in current T2I models.
2. It doesn't affect the performance in our experiments~
